examples/all-clusters-app/linux/main-common.cpp

```diff
@@ -50,297 +50,43 @@
-    {
-        GeneralFaults<kMaxNetworkFaults> previous;
-        GeneralFaults<kMaxNetworkFaults> current;
-
-#if CHIP_CONFIG_TEST
-        // On Linux Simulation, set following radio faults statically.
-        ReturnOnFailure(previous.add(EMBER_ZCL_NETWORK_FAULT_TYPE_HARDWARE_FAILURE));
-        ReturnOnFailure(previous.add(EMBER_ZCL_NETWORK_FAULT_TYPE_NETWORK_JAMMED));
-
-        ReturnOnFailure(current.add(EMBER_ZCL_NETWORK_FAULT_TYPE_HARDWARE_FAILURE));
-        ReturnOnFailure(current.add(EMBER_ZCL_NETWORK_FAULT_TYPE_NETWORK_JAMMED));
-        ReturnOnFailure(current.add(EMBER_ZCL_NETWORK_FAULT_TYPE_CONNECTION_FAILED));
-#endif
-        Clusters::GeneralDiagnosticsServer::Instance().OnNetworkFaultsDetect(previous, current);
-    }
-    else
-    {
-        ChipLogError(DeviceLayer, "Unknow event ID:%d", eventId);
-    }
-}
-
-/**
- * Should be called when a switch operation takes place on the Node.
- */
-void HandleSwitchEvent(intptr_t arg)
-{
-    uint32_t eventId = static_cast<uint32_t>(arg);
-
-    EndpointId endpoint      = 1;
-    uint8_t newPosition      = 20;
-    uint8_t previousPosition = 10;
-    uint8_t count            = 3;
-
-    if (eventId == Clusters::Switch::Events::SwitchLatched::Id)
-    {
-        Clusters::SwitchServer::Instance().OnSwitchLatch(endpoint, newPosition);
-    }
-    else if (eventId == Clusters::Switch::Events::InitialPress::Id)
-    {
-        Clusters::SwitchServer::Instance().OnInitialPress(endpoint, newPosition);
-    }
-    else if (eventId == Clusters::Switch::Events::LongPress::Id)
-    {
-        Clusters::SwitchServer::Instance().OnLongPress(endpoint, newPosition);
-    }
-    else if (eventId == Clusters::Switch::Events::ShortRelease::Id)
-    {
-        Clusters::SwitchServer::Instance().OnShortRelease(endpoint, previousPosition);
-    }
-    else if (eventId == Clusters::Switch::Events::LongRelease::Id)
-    {
-        Clusters::SwitchServer::Instance().OnLongRelease(endpoint, previousPosition);
-    }
-    else if (eventId == Clusters::Switch::Events::MultiPressOngoing::Id)
-    {
-        Clusters::SwitchServer::Instance().OnMultiPressOngoing(endpoint, newPosition, count);
-    }
-    else if (eventId == Clusters::Switch::Events::MultiPressComplete::Id)
-    {
-        Clusters::SwitchServer::Instance().OnMultiPressComplete(endpoint, newPosition, count);
-    }
-    else
-    {
-        ChipLogError(DeviceLayer, "Unknow event ID:%d", eventId);
-    }
-}
-
-// when the shell is enabled, don't intercept signals since it prevents the user from
-// using expected commands like CTRL-C to quit the application. (see issue #17845)
-// We should stop using signals for those faults, and move to a different notification
-// means, like a pipe. (see issue #19114)
-#if !defined(ENABLE_CHIP_SHELL)
-void OnRebootSignalHandler(int signum)
+void * EventCommandListenerTask(void * arg)
 {
-    ChipLogDetail(DeviceLayer, "Caught signal %d", signum);
+    int fd;
```

Comment:
why is this declared outside the for loop?

How about adding a condition to stop the infinite loop as well? How does this thread complete?

Reply:
I added the condition to exit from the thread in case failing to open the fifo, otherwise, the thread is cancel by the chip main thread within ApplicationExit()